src/metabase/models/card.clj

```diff
@@ -177,6 +177,26 @@
   (cond-> card
     (seq (:dataset_query card)) (update :dataset_query mbql.normalize/normalize)))
 
+(defn template-tag-parameters
+  "Transforms native query's `template-tags` into `parameters`.
+  An older style was to not include `:template-tags` onto cards as parameters. I think this is a mistake and they should always be there. Apparently lots of e2e tests are sloppy about this so this is included as a convenience."
+  [card]
+  ;; NOTE: this should mirror `getTemplateTagParameters` in frontend/src/metabase/parameters/utils/cards.js
+  (for [[_ {tag-type :type, widget-type :widget-type, :as tag}] (get-in card [:dataset_query :native :template-tags])
+        :when                         (and tag-type
+                                           (or widget-type (not= tag-type :dimension)))]
+    {:id      (:id tag)
+     :type    (or widget-type (cond (= tag-type :date)   :date/single
+                                    (= tag-type :string) :string/=
+                                    (= tag-type :number) :number/=
+                                    :else                :category))
+     :target  (if (= tag-type :dimension)
+                [:dimension [:template-tag (:name tag)]]
+                [:variable  [:template-tag (:name tag)]])
+     :name    (:display-name tag)
+     :slug    (:name tag)
```

Comment:
This should be updated to include `values_query_type`, `values_source_type`, and `values_source_config` as here https://github.com/metabase/metabase/blob/0a3708fb761422938aabc449e976145b3d585c59/frontend/src/metabase-lib/parameters/utils/template-tags.ts#L33

Also the schema for `TemplateTag` should be properly updated.

Reply:
I don't believe that's possible. On a dashboard a single parameter could point to many targets. We really want this information to live on a paramater and not update these template tags. Imagine a dashboard with 15 cards, each of which had a template tag pointing to the same field, but each of which had different static lists, different cards, etc for the template tag.

This is just not a concern of a template tag and does not belong here.

We _really_ should only put this information on a single parameter.